src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js

```diff
@@ -288,67 +258,82 @@ class PopoverReportActionContextMenu extends React.Component {
-                    shouldSetModalVisibility={this.state.shouldSetModalVisibilityForDeleteConfirmation}
-                    onConfirm={this.confirmDeleteAndHideModal}
-                    onCancel={this.hideDeleteModal}
-                    onModalHide={this.callbackWhenDeleteModalHide}
-                    prompt={this.props.translate('reportActionContextMenu.deleteConfirmation', {action: this.state.reportAction})}
-                    confirmText={this.props.translate('common.delete')}
-                    cancelText={this.props.translate('common.cancel')}
-                    danger
+    const showDeleteModal = (reportID, reportAction, shouldSetModalVisibility = true, onConfirm = () => {}, onCancel = () => {}) => {
+        onCancelDeleteModal.current = onCancel;
+        onComfirmDeleteModal.current = onConfirm;
+
+        reportIDRef.current = reportID;
+        reportActionRef.current = reportAction;
+
+        setShouldSetModalVisibilityForDeleteConfirmation(shouldSetModalVisibility);
+        setIsDeleteCommentConfirmModalVisible(true);
+    };
+
+    useImperativeHandle(ref, () => ({
+        showContextMenu,
+        hideContextMenu,
+        showDeleteModal,
+        hideDeleteModal,
+        isActiveReportAction,
+        instanceID,
+        runAndResetOnPopoverHide,
+        clearActiveReportAction,
+    }));
+
+    const reportAction = reportActionRef.current;
+
+    return (
+        <>
+            <PopoverWithMeasuredContent
+                isVisible={isPopoverVisible}
+                onClose={hideContextMenu}
+                onModalShow={runAndResetOnPopoverShow}
+                onModalHide={runAndResetOnPopoverHide}
+                anchorPosition={popoverAnchorPosition.current}
+                animationIn="fadeIn"
+                disableAnimation={false}
+                animationOutTiming={1}
+                shouldSetModalVisibility={false}
+                fullscreen
+                withoutOverlay
+                anchorRef={anchorRef}
+            >
+                <BaseReportActionContextMenu
+                    isVisible
+                    type={typeRef.current}
+                    reportID={reportIDRef.current}
+                    reportActionID={reportActionIDRef.current}
+                    draftMessage={reportActionDraftMessageRef.current}
+                    selection={selectionRef.current}
+                    isArchivedRoom={isRoomArchived}
+                    isChronosReport={isChronosReportEnabled}
+                    isPinnedChat={isChatPinned}
+                    isUnreadChat={hasUnreadMessages}
+                    anchor={contextMenuTargetNode}
+                    contentRef={contentRef}
+                    originalReportID={originalReportIDRef.current}
                 />
-            </>
-        );
-    }
+            </PopoverWithMeasuredContent>
+            <ConfirmModal
+                title={translate('reportActionContextMenu.deleteAction', {reportAction})}
```

Comment:
@OlimpiaZurek Why you change this line? It should be
`title={this.props.translate('reportActionContextMenu.deleteAction', {action: this.state.reportAction})}`

It causes regression [here](https://github.com/Expensify/App/issues/28861)